# -*- coding: utf-8 -*-
"""
exception_01.py
"""

# 예외처리
# 에러 : 잘못된 것(프로그램의 실행이 안됨)
# 예외 : 잘못된 것
# (프로그램의 실행은 되지만...
# 특정 상황, 조건이 만족될 때 에러가 발생)
# 예외처리 : 예외가 발생한 경우 프로그램이 
# 강제로 종료되는 현상이 발생합니다.
# 예외처리는 예외가 발생하더라도
# 프로그램이 강제 종료되는 현상을 
# 방지할 수 있는 문법입니다

s1 = "Hello "
s2 = "Python~!"
r = s1 + s2
print(r)

# 파이썬은 문자열 결합연산은
# 문자열 사이에서만 가능하기 때문에
# 아래의 코드는 예외가 발생
#s1 = 15
#r = s1 + s2
#print(r)

numbers = list(range(1,11))

index = int(input("출력할 인덱스 입력 : "))

# 정해진 인덱스의 범위를 벗어난 수가 입력되면
# 예외가 발생
print(numbers[index])

# 예외가 발생한 경우 아래의 실행코드는
# 무시됩니다.
print('프로그램 종료')


n1 = int(input("첫번째 정수 입력 : "))
n2 = int(input("두번째 정수 입력 : "))
r = n1 / n2
print(f'{n1} / {n2} = {r}')
print('프로그램 종료')











